src/map/pc.cpp

```diff
@@ -1926,6 +1926,11 @@ bool pc_lastpoint_special( map_session_data& sd ){
 		return true;
 	}
 
+	if (strcmpi(sd.status.last_point.map, MAP_JAIL) == 0) {
```

Comment:
Doesnt that make normal "jail visitors" stuck on that map too?

Reply:
It shouldn't. Jail has the nosave mapflag so the player shouldn't have their lastpoint set to the jail map.

Reply:
Unfortunately, I don't think there is another way